internal/util/kafka.go

```diff
@@ -65,6 +64,85 @@ func (k kafkautil) NewWriter(topic string) *kafka.Writer {
 	}
 }
 
+func getSaslAndTLSConfig() (sasl.Mechanism, *tls.Config) {
+	username := config.DefaultConfig.KafkaUsername
+	password := config.DefaultConfig.KafkaPassword
+	saslmechanismName := config.DefaultConfig.KafkaSaslMechanism
+
+	tlsConfig, err := createTLSConfig(config.DefaultConfig.KafkaCAPath)
+	if err != nil {
+		log.Error().Err(err).Msg("Error creating TLS configuration for Kafka:")
+		tlsConfig = &tls.Config{} // Providing default empty TLS configuration
+	}
+
+	saslMechanism, err := createSaslMechanism(
+		saslmechanismName,
+		username,
+		password,
+	)
+	if err != nil {
+		log.Error().Err(err).Msg("Error creating SASL Mechanism for Kafka, using plain mechanism")
+	}
+
+	return saslMechanism, tlsConfig
+}
+
+func createTLSConfig(pathToCert string) (*tls.Config, error) {
+
+	tlsConfig := tls.Config{
+		MinVersion: tls.VersionTLS12,
```

Comment:
```suggestion
		MinVersion: tls.VersionTLS12,
		InsecureSkipVerify: true,
```
I get below error if we do not add [InsecureSkipVerify](https://pkg.go.dev/crypto/tls#:~:text=VerifyConnection%20or%20VerifyPeerCertificate.-,InsecureSkipVerify,-bool%0A%0A%09//%20CipherSuites%20is)
```
{"level":"error","error":"fetching message: failed to dial: tls: failed to verify certificate: x509: certificate relies on legacy Common Name field, use SANs instead","time":"2024-03-17T18:24:25+05:30","caller":"/Users/suraj/redhat-projects/tmp/config-manager/internal/cmd/inventoryconsumer/cmd.go:33","message":"unable to read message"}
```
In our other service [ros-ocp-backend](https://github.com/RedHatInsights/ros-ocp-backend/blob/main/internal/kafka/consumer.go#L32-L50) we do not need to explicitly add any config to ignore server side certification check. Therefore I assume in confluent-kafka-go by default we do not verify server side certificate.